internal/config/identity/ldap/ldap.go

```diff
@@ -354,3 +403,99 @@ func (l *Config) LookupGroupMemberships(userDistNames []string, userDNToUsername
 
 	return res, nil
 }
+
+// QuickNormalizeDN - normalizes the given DN without checking if it is valid or
+// exists in the LDAP directory. Returns input if error
+func (l Config) QuickNormalizeDN(dn string) string {
+	if normDN, err := xldap.NormalizeDN(dn); err == nil {
+		return normDN
+	}
+	return dn
+}
+
+// DecodeDN - denormalizes the given DN by unescaping any escaped characters.
+// Returns input if error
+func (l Config) DecodeDN(dn string) string {
+	if denormDN, err := decodeDN(dn); err == nil {
+		return denormDN
+	}
+	return dn
+}
+
+// Remove leading and trailing spaces from the attribute type and value
+// and unescape any escaped characters in these fields
+//
+// decodeDN is pulled from the go-ldap library
+// https://github.com/go-ldap/ldap/blob/dbdc485259442f987d83e604cd4f5859cfc1be58/dn.go
+func decodeDN(str string) (string, error) {
```

Comment:
why are doing this here?

Reply:
This is to decode the internal unicode representation of some characters into human-readable characters when they are being sent externally (i.e. `cn=\d0\98\d0\b2\d0\b0\d0\bd\d0\be\d0\b2 \d0\98\d0\b2\d0\b0\d0\bd,cn=Users,dc=min,dc=io` is converted to `cn=Иванов Иван,cn=Users,dc=min,dc=io`)

Reply:
Can we move this to `pkg/ldap` and also add some good amount of unit tests, we can't just fork such sensitive code without unit tests for this function.

We should also keep the code as `pkg/ldap/decode_dn_contrib.go` where this file carries an external license header as part of the source file - since its just a copy from the upstream project.

if its a new implementation fresh without copying from upstream then we are good.